tests/eval_files/trio232.py

```diff
@@ -0,0 +1,135 @@
+import io
+from io import BufferedRandom, BufferedReader, BufferedWriter, TextIOWrapper
+from typing import Any, Optional
+
+blah: Any = None
+
+
+async def file_text(f: io.TextIOWrapper):
+    f.read()  # TRIO232: 4, 'read', 'f'
+
+
+async def file_binary_read(f: io.BufferedReader):
+    f.read()  # TRIO232: 4, 'read', 'f'
+
+
+async def file_binary_write(f: io.BufferedWriter):
+    f.read()  # TRIO232: 4, 'read', 'f'
+
+
+async def file_binary_readwrite(f: io.BufferedRandom):
+    f.read()  # TRIO232: 4, 'read', 'f'
+
+
+async def file_text_(f: TextIOWrapper):
+    f.read()  # TRIO232: 4, 'read', 'f'
+
+
+async def file_binary_read_(f: BufferedReader):
+    f.read()  # TRIO232: 4, 'read', 'f'
+
+
+async def file_binary_write_(f: BufferedWriter):
+    f.read()  # TRIO232: 4, 'read', 'f'
+
+
+async def file_binary_readwrite_(f: BufferedRandom):
+    f.read()  # TRIO232: 4, 'read', 'f'
+
+
+async def file_text_3(f: TextIOWrapper = blah):
+    f.read()  # TRIO232: 4, 'read', 'f'
+
+
+async def file_text_4(f: TextIOWrapper | None):
+    if f:
+        f.read()  # TRIO232: 8, 'read', 'f'
```

Comment:
Would this still give a lint error without the `if`, or in an `else`?  It'd be nice to demonstrate that with tests.

Reply:
Yes, it currently does not care at all that it could be a None - but I had type-checking turned on in my editor when writing this one hence the `if`'s. You're completely right it should be shown in tests though - and will fix in followup PR :+1: